src/Features/CSharp/Portable/InlineParameterNameHints/InlineParamNameHintsService.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.InlineParameterNameHints;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.InlineParameterNameHints
+{
+    /// <summary>
+    /// The service to locate the positions in which the adornments should appear
+    /// as well as associate the adornments back to the parameter name
+    /// </summary>
+    [ExportLanguageService(typeof(IInlineParamNameHintsService), LanguageNames.CSharp), Shared]
+    internal class InlineParamNameHintsService : IInlineParamNameHintsService
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public InlineParamNameHintsService()
+        {
+        }
+
+        public async Task<IEnumerable<NameAndSpan>> GetInlineParameterNameHintsAsync(
+            Document document,
+            TextSpan textSpan,
+            CancellationToken cancellationToken)
+        {
+            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var node = await tree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+            var spans = new List<NameAndSpan>();
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var invocations = node.DescendantNodes().OfType<InvocationExpressionSyntax>();
+
+            foreach (var invo in invocations)
```

Comment:
```suggestion
            foreach (var invocation in invocations)
```

We've got completion, characters are cheap. :-)